core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorResponseFactory.java

```diff
@@ -27,28 +24,19 @@
 import com.github.mizool.core.exception.RuleViolationException;
 
 @Slf4j
-public class ErrorHandler
+public class ErrorResponseFactory
 {
-    private final ErrorHandlingCatalog errorHandlingCatalog;
-    private final ErrorMapper errorMapper;
     private final ConstraintViolationMapper constraintViolationMapper;
     private final RuleViolationMapper ruleViolationMapper;
+    private final ClientErrorMapper clientErrorMapper;
+    private final GenericErrorMapper genericErrorMapper;
 
-    public ErrorHandler()
-    {
-        errorHandlingCatalog = new ErrorHandlingCatalog();
-        errorMapper = new ErrorMapper();
-        constraintViolationMapper = new ConstraintViolationMapper();
-        ruleViolationMapper = new RuleViolationMapper();
-    }
-
-    @Inject
-    protected ErrorHandler(ErrorHandlingCatalog errorHandlingCatalog)
+    public ErrorResponseFactory()
```

Comment:
We no longer need a constructor; move the assignments to the field declarations.